### PR TITLE
Support for financial year customisation

### PR DESCRIPTION
**Pull request by sausin**
Instead of the current default financial year start from 01 Jan, this allows for customized start of the financial year, for eg. from 01 April (in case of India).

### CODE REVIEW

File: app/Http/Controllers/Common/Dashboard.php [hunk 1]
```diff
@@ -33,6 +37,7 @@ class Dashboard extends Controller
     public function index()
     {
         $this->today = Date::today();
+        $this->financial_start = $financial_start = Date::parse(Setting::where('key', 'general.financial_start')->first()->value)->format('Y-m-d');
```
Comment (denisdulici) [inline app/Http/Controllers/Common/Dashboard.php]: Find: `Setting::where('key', 'general.financial_start')->first()->value`
Replace: `setting('general.financial_start')`

File: app/Http/Controllers/Common/Dashboard.php [hunk 2]
```diff
@@ -55,7 +60,8 @@ public function index()
             'donut_expenses',
             'accounts',
             'latest_incomes',
-            'latest_expenses'
+            'latest_expenses',
+            'financial_start'
```
Comment (denisdulici) [inline app/Http/Controllers/Common/Dashboard.php]: This is also not needed as you can use the `setting` helper in views too.

Comment (sausin) [reply app/Http/Controllers/Common/Dashboard.php]: Thanks for your suggestions! I'll update the code to make it neater.

I didn't look at the reports, so checking those now.

File: app/Http/Controllers/Common/Dashboard.php (hunk 1 above)
Comment (sausin) [reply app/Http/Controllers/Common/Dashboard.php]: Will update to use the helper

**Comment by sausin** [comment]
@denisdulici please check now

File: database/seeds/Settings.php [hunk 3]
```diff
@@ -28,6 +28,7 @@ private function create()
 
         Setting::set([
             'general.date_format'               => 'd M Y',
+            'general.financial_start'           => '01 Jan',
```
Comment (denisdulici) [inline database/seeds/Settings.php]: What happens for other languages?

**Comment by sausin** [comment]
Using Date function now so it should be fixed. Let me know if anything else

**Comment by denisdulici** [comment]
I get the following error when I change the fiscal start (i.e. 01 April) and try to access any report.

screenshot_1545487935

**Comment by sausin** [comment]
@denisdulici Thanks for checking and sorry I missed a change in the report files. It should be good now.

I assume `Tax Summary` and `Profit & Loss` are working properly for you without the missing modifications.

**Comment by denisdulici** [comment]
Everything seems to be working fine now, thanks for your contribution :clap:

**Comment by sausin** [comment]
@denisdulici glad to be of help :+1:

**Pull request merged**